go/libraries/doltcore/env/actions/docs.go

```diff
@@ -179,5 +189,26 @@ func SaveDocsFromWorkingExcludingFSChanges(ctx context.Context, dEnv *env.DoltEn
 		docsToSave = dEnv.Docs
 	}
 
-	return SaveTrackedDocs(ctx, dEnv, workingRoot, workingRoot, docsToSave)
+	return SaveTrackedDocs(ctx, dEnv.DocsReadWriter(), workingRoot, workingRoot, docsToSave)
+}
+
+// GetTablesOrDocs takes a slice of table or file names. Table names are returned as given. Supported doc names are
+// read from disk and their name replace with the names of the dolt_docs system table in the input slice. Supported docs
+// are returned in the second return param.
+func GetTablesOrDocs(drw env.DocsReadWriter, tablesOrFiles []string) (tables []string, docs doltdocs.Docs, err error) {
+	for i, tbl := range tablesOrFiles {
+		if _, ok := doltdocs.IsSupportedDoc(tbl); ok {
+			docAr, err := drw.GetDocsOnDisk(tbl)
+			if err != nil {
+				return nil, nil, err
+			}
+			doc := docAr[0]
```

Comment:
Is there a better way to do this?

Reply:
Not really, but you should do a bound check